test/testutilsio.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2023  Mike Tzou (Chocobo1)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include <QtGlobal>
+#include <QTest>
+
+#include "base/global.h"
+#include "base/path.h"
+#include "base/utils/io.h"
+
+class TestUtilsIO final : public QObject
+{
+    Q_OBJECT
+    Q_DISABLE_COPY_MOVE(TestUtilsIO)
+
+public:
+    TestUtilsIO() = default;
+
+private slots:
+    void testReadFile() const
```

Comment:
Don't know what you mean, the CI passed these tests.

Reply:
> Don't know what you mean

E.g. what do you expect to check with the following test?
```c++
QCOMPARE(Utils::IO::readFile(testBinary, (100 * 1024 * 1024)).has_value(), true);
```

Reply:
I want to read an existing file. It was a bit puzzling for me to manage the working directory path. So I decided to read its binary.

Reply:
You didn't understand my question.
If this test fails, what is it supposed to mean?
I believe the tests are intended to check the correctness of logic, isn't that right? But this test may fail for some other reasons, for example, due to some I/O errors.

Reply:
So how would you do?

Reply:
> So how would you do it?

I would just not add tests that can give ambiguous results.

Reply:
>I would just not add tests that can give ambiguous results.

I fail to see why it returns ambiguous results... and you choose to not test the 'read successful' path which I find absurd.

Reply:
> I fail to see why it returns ambiguous results...

Isn't it obvious? Moreover, I have already explained it above. If your test fails, it's not necessarily because of an error in the tested algorithm.

What do you want to check with this test? Just that some I/O operation succeeded/failed in some particular case? And why, excuse me, is this necessary?Do you want to test the I/O subsystem on the target machine? Or the correctness of the algorithms?

Reply:
If you never test the successful case, how can you have confidence that it will work as expected?

Reply:
Is it better to have something unconfirmed or confirmed in an unreliable way?
Using your words, I fail to see how can you have confidence that it will work as expected. Even without taking into account unrelated errors (e.g. I/O errors), your test does not have strict preconditions to check any particular expectation.
What would I expect from tested algorithm? Having a limit of N bytes, it must read a file from 0 to N bytes in size, and it must fail to read a file larger than N bytes. So the test should necessarily check the extreme conditions, i.e. reading a file of size N bytes and N+1 bytes. But there is only reading a file of unknown size...

Reply:
>What would I expect from tested algorithm? Having a limit of N bytes, it must read a file from 0 to N bytes in size, and it must fail to read a file larger than N bytes. So the test should necessarily check the extreme conditions, i.e. reading a file of size N bytes and N+1 bytes. But there is only reading a file of unknown size...

This is now addressed.